Add release notes for 14 July 2025 (#539)
catalyst-cloud/catalystcloud-docs: source/compute/gpu-support.rst

GPU Support in Virtual Servers
##############################

********
Overview
********

Catalyst Cloud provides several options for deloying instances with
NVIDIA GPU acceleration that can be utilised by selecting the desired GPU
:ref:`flavor <instance-types>` when creating the instances.

All GPU instances require suitable drivers to be installed before the GPU can
be utilised. Catalyst Cloud is not permitted to provide modified operating
system images so customers deploying GPU instances will need to install
suitable drivers themselves.

Instances built using C2-GPU flavors use virtual GPU partitions which have
special driver reqirements. Standard NVIDIA drivers supplied with operating
systems cannot be used for C2-GPU. Please refer to the
:ref:`C2-GPU instructions<c2-gpu-support>` for details.

All other GPU instances use direct pass-through of the entire GPU hardware,
so any driver that is compatible with the GPU hardware can be installed.
Catalyst Cloud recommends using drivers supplied by the operating system vendor
or downloaded directly from NVIDIA.

> These steps do not apply to Kubernetes worker nodes. Fo rmore information

Reviewer: ```suggestion
These steps do not apply to Kubernetes worker nodes. For more information
```

        If that doesn't work, try running ``sudo modprobe nvidia`` to ensure the NVIDIA
        driver is loaded, or reboot the instance.

        Please refer to the :ref:`CUDA section <nvidia-cuda-support>` for instructions on installing the
        CUDA toolkit.

    .. tab:: Microsoft Windows

        To use GPUs in Windows, simply download and install the latest drivers from:

        https://www.nvidia.com/en-us/drivers/

        Select the GPU model according to the flavour being used:

        .. list-table::
           :header-rows: 1

           * - Compute Flavour
             - GPU Model
           * - C1A-GPU
             - NVIDIA RTX A6000
           * - C3-GPU
             - NVIDIA L40S

        Once the driver package is installed, verify that it is installed by checking
        the GPU state in Device Manager, or running ``nvidia-smi`` from the command
        prompt.

Automated Driver Installation
=============================

For a more streamlined setup of GPU instances, the necessary GPU driver packages
can be installed via :ref:`user data when creating instances <instance_initialisation>`.

> This means the GPU is ready use use within a few minutes of the instance booting

Reviewer: ```suggestion
This means the GPU is ready to use within a few minutes of the instance booting
```

    sudo tee /etc/ld.so.conf.d/cuda.conf <<< /usr/local/cuda/lib64
    sudo ldconfig

.. _nvidia-cuda-support:

************
CUDA Support
************

> The CUDA version supported is specifiec to the NVIDIA driver release version.

Reviewer: ```suggestion
The CUDA version supported is specific to the NVIDIA driver release version.
```

For most GPU flavors, simply download the appropriate CUDA toolkit version from
NVIDIA to match the driver release used in the instance:

https://docs.nvidia.com/cuda/cuda-installation-guide-linux/

Some operating systems (e.g. Ubuntu) include CUDA packages in their
repositories that can also be used instead, although they are usually older
versions.

In the case of C2-GPU instances the CUDA toolkit version is currently limited
by driver release 535 which officially supports CUDA 12.2.

> NVIDIA provide compatbility libraries to allow applications compiled against

Reviewer: ```suggestion
NVIDIA provide compatibility libraries to allow applications compiled against
```
